src/System.Linq.Parallel/tests/DegreeOfParallelismTests.cs

```diff
@@ -1,140 +1,86 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Xunit;
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Xunit;
 
 namespace Test
 {
     public class DegreeOfParallelismTests
     {
-        [Fact]
-        [OuterLoop]
-        public static void RunDopTests()
+        public static IEnumerable<object[]> DegreeData(object[] counts, object[] degrees)
         {
-            RunDopTest(1, false);
-            RunDopTest(4, false);
-            RunDopTest(512, false);
-            RunDopTest(513, true);
+            foreach (object[] results in UnorderedSources.Ranges(counts.Cast<int>()))
+            {
+                foreach (int degree in degrees.DefaultIfEmpty(results[1]).Cast<int>())
+                {
+                    yield return new object[] { results[0], results[1], degree };
+                }
+            }
         }
 
-        [Fact]
+        [Theory]
+        [MemberData("DegreeData", new[] { 1024 }, new[] { 1, 4, 512 })]
         [OuterLoop]
-        public static void RunDopBarrierTest()
+        public static void DegreeOfParallelism(Labelled<ParallelQuery<int>> labelled, int count, int degree)
         {
-            RunDopBarrierTestCore(1);
-            RunDopBarrierTestCore(4);
-            RunDopBarrierTestCore(32);
-            // These variations take way too long and hit the 100s timeout.
-            //RunDopBarrierTest(64);
-            //RunDopBarrierTest(512);
+            Assert.Equal(Functions.SumRange(0, count), labelled.Item.WithDegreeOfParallelism(degree).Sum());
         }
 
-        [Fact]
+        [Theory]
+        [MemberData("DegreeData", new[] { 1, 4, 32 }, new object[] { })]
         [OuterLoop]
-        public static void RunDopPipeliningTest()
+        public static void DegreeOfParallelism_Barrier(Labelled<ParallelQuery<int>> labelled, int count, int degree)
         {
-            RunDopPipeliningTestCore(1000000, 1);
-            RunDopPipeliningTestCore(1000000, 4);
-            RunDopPipeliningTestCore(1000000, 64);
-            RunDopPipeliningTestCore(1000000, 512);
+            //int minWorker, minAsyncIO;
+            //ThreadPool.GetMinThreads(out minWorker, out minAsyncIO);
+            //ThreadPool.SetMinThreads(degree, minAsyncIO);
```

Comment:
Why is this commented out?  You should consider reinstating it, along with code to reset the minimums back to what they were if you changed them due to them being too low.  If you don't, it could take at least seconds for the thread pool to otherwise ramp up to 32 threads, and it could take minutes for it to ramp up to 512 threads, given that it'd likely only be doing so based on its starvation policy which will only inject 1 or 2 threads per second if it sees things stalled.